unit_test/src/heredoc_cases.c

```diff
@@ -0,0 +1,65 @@
+#include <unit_test.h>
+
+/*******************************************************************************/
+/*                            heredoc_basic_cases                              */
+/*******************************************************************************/
+
+void	assert_heredoc_basic(char *delimiter, int32_t expected_output)
+{
+	int32_t	output_fd;
+
+	output_fd = here_doc(delimiter);
+	cr_assert_eq(output_fd, expected_output);
+}
+
+Test(heredoc, null_delimiter)
+{
+	assert_heredoc_basic(NULL, ERROR);
+}
+
+Test(heredoc, empty_delimiter)
+{
+	assert_heredoc_basic("", ERROR);
+}
+
+/*******************************************************************************/
+/*                               heredoc_cases                                 */
+/*******************************************************************************/
+
+void	assert_heredoc(char *delimiter, char *input, char *expected_output)
+{
+	int		pipe_fd[2];
+	int		output_fd;
+	char 	buf[1024];
+	ssize_t	num_bytes;
+
+	if (pipe(pipe_fd) == -1) {
+		perror("pipe");
+		exit(EXIT_FAILURE);
```

Comment:
normally you never want to call exit but want to use an assertion e.g. cr_asssert(pipe == SUCCESS, "%s", perror("pipe"));